source/.clang-format

```diff
@@ -0,0 +1,86 @@
+---
```

Comment:
Add this file in it's own commit please.

Reply:
I'll separate it at the very end after we're done with this and good to merge.